types/styled-components/test/index.tsx

```diff
@@ -1151,8 +1151,7 @@ const wrapperClassNoChildren = <StyledWrapperClassFuncChild>Text</StyledWrapperC
 
 const WrapperFunction: React.FunctionComponent<WrapperProps> = () => <div />;
 const StyledWrapperFunction = styled(WrapperFunction)``;
-// React.FunctionComponent typings always add `children` to props, so this should accept children
-const wrapperFunction = <StyledWrapperFunction>Text</StyledWrapperFunction>;
+const wrapperFunction = <StyledWrapperFunction />;
```

Comment:
why did this test have to change? wasn't it testing if `StyledWrapperFunction` component accepted children? is that no longer the case?

Reply:
It's relying on deprecated features of `React.FunctionComponent`. `React.FunctionComponent` currently implicitly adds `children` to the accepted props which we intend to remove in React 18.

So this test was testing features of `@types/react` not `@types/styled-components`.

The non-deprecated behavior is tested just below this test:

```tsx
const WrapperFunc = (props: WrapperProps) => <div />;
const StyledWrapperFunc = styled(WrapperFunc)``;
// No `children` in props, so this should generate an error
const wrapperFunc = <StyledWrapperFunc>Text</StyledWrapperFunc>; // $ExpectError
```